File: p1/p1.cs
class P1{
  static int SumDivisible(int n, int max){
    int p = max / n;
    return n * (p * (p+1)) / 2;
  }
  static void Main(string[] args){
    int max = 999;
    if (args.Length >= 1){
      max = int.Parse(args[0]) - 1;
    }
    int sum = SumDivisible(3, max) + SumDivisible(5, max) - SumDivisible(15, max);
    System.Console.Write("Sum of the multipes of 3 & 5 less than {0}: {1}\n", max + 1, sum);
  }
}

File: p1/p1.rs
use std::env;

fn sum_divisible(n: i32, max: i32) -> i32 {
    let p = max / n;
    n * (p * (p + 1)) / 2
}

fn main() {
    let mut max: i32 = 999;
    let args: Vec<_> = env::args().collect();
    if args.len() > 1 {
        max = args[1].parse().unwrap();
        max = max - 1;
    }
    let sum = sum_divisible(3, max) + sum_divisible(5, max) - sum_divisible(15, max);
    println!("Sum of the multiples of 3 & 5 less than {}: {}", max + 1, sum);
}
